Lib/ufo2ft/util.py

```diff
@@ -528,3 +530,19 @@ def prune_unknown_kwargs(kwargs, *callables):
     for func in callables:
         known_args.update(getfullargspec(func).args)
     return {k: v for k, v in kwargs.items() if k in known_args}
+
+
+def ensure_all_sources_have_names(doc: DesignSpaceDocument) -> None:
+    """Change in-place the given document to make sure that all <source> elements
+    have a unique name assigned.
+    """
+    used_names: Set[str] = set()
+    counter = 1
+    for source in doc.sources:
+        while (
+            source.name is None
+            or "temp_master" in source.name  # Legacy quirk in designspaceLib
+            or source.name in used_names
+        ):
+            source.name = f"source.{counter}"
```

Comment:
you may as well use `temp_master.%d` as the prefix.

Reply:
Also, @belluzj use_names is never added to?!

Reply:
thanks for spotting that

Reply:
Initially I was not using "temp_master" because designspaceLib doesn't write out source names if they're called "temp_master", and because we were using serialize and deserialize again to deep-copy, it meant that these synthetic names were lost when deep-copying. So you're right it's not an issue anymore to use temp_master as a default source name, because the new deepcopy is not using XML anymore so it will preserve them.

https://github.com/fonttools/fonttools/blob/main/Lib/fontTools/designspaceLib/__init__.py#L1618-L1620